tools/devworkspace-generator/src/devfile/dev-container-component-finder.ts

```diff
@@ -10,14 +10,26 @@
 
 import { DevfileContext } from '../api/devfile-context';
 import { V1alpha2DevWorkspaceSpecTemplateComponents } from '@devfile/api';
-import { injectable } from 'inversify';
+import { inject, injectable } from 'inversify';
+import { DevContainerComponentInserter } from './dev-container-component-inserter';
 
 /**
  * Need to find dev container from main dev workspace
  */
 @injectable()
 export class DevContainerComponentFinder {
-  async find(devfileContext: DevfileContext): Promise<V1alpha2DevWorkspaceSpecTemplateComponents | undefined> {
+  @inject(DevContainerComponentInserter)
+  private devContainerComponentInserter: DevContainerComponentInserter;
+
+  async find(
+    devfileContext: DevfileContext,
+    injectDefaultComponent?: string,
+    defaultComponentImage?: string
+  ): Promise<V1alpha2DevWorkspaceSpecTemplateComponents | undefined> {
+    // if a devfile contains a parent, we should not add a default dev container
+    if (devfileContext.devfile?.parent) {
```

Comment:
Is it possible to have the devfileContext without a devfile?

Reply:
it's mandatory field, it should be defined in all cases

Reply:
```
if (devfileContext.devfile?.parent) {
...
```
->
```
if (devfileContext.devfile.parent) {
...
```
Can you change it?